### PR TITLE
docs: clarify the meaning of cleanup policy of ceph cluster crd

### PR DESCRIPTION
**Pull request by satoru-takeuchi**
**Description of your changes:**

The description of cleanupPolicy is a bit ambiguous. First, we should clarify that both skipping orchestration and data cleanup will happen iff "yes-really-destroy-data" is set to `confirmation`. Second, We should also describe wiping devices will happen even if user don't specify `sanitizeDisks` explicitly.

**Which issue is resolved by this Pull Request:**

Nothing

**Checklist:**

- [o] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [o] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [o] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [o] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [o] Documentation has been updated, if necessary.
- [o] Unit tests have been added, if necessary.
- [o] Integration tests have been added, if necessary.
- [o] Pending release notes updated with breaking and/or notable changes, if necessary.
- [o] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [o] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

File: cluster/examples/kubernetes/ceph/cluster.yaml [hunk 1]
```diff
@@ -89,12 +89,13 @@ spec:
...
   crashCollector:
     disable: false
+  # automate [data cleanup process](https://github.com/rook/rook/blob/master/Documentation/ceph-teardown.md#delete-the-data-on-hosts) in cluster destruction.
   cleanupPolicy:
-    # cleanup should only be added to the cluster when the cluster is about to be deleted.
-    # After any field of the cleanup policy is set, Rook will stop configuring the cluster as if the cluster is about
-    # to be destroyed in order to prevent these settings from being deployed unintentionally.
-    # To signify that automatic deletion is desired, use the value "yes-really-destroy-data". Only this and an empty
-    # string are valid values for this field.
+    # confirmation indicates whether user really want the automatic cleanup. Only empty string(default)
```
Comment (leseb) [inline cluster/examples/kubernetes/ceph/cluster.yaml:94]:
```suggestion
    # confirmation indicates whether user really wants the automatic cleanup. Only empty string(default)
```

**Review by leseb — CHANGES_REQUESTED** [review]
nit

**Review by travisn — CHANGES_REQUESTED** [review]
I had too many suggestions so I created a new commit to consider, this is a very complicated topic. How about [these suggestions](https://github.com/travisn/rook/commit/594e3789ee99ad5991050de31b8847c9ca74ec4f)?

File: cluster/examples/kubernetes/ceph/cluster.yaml (hunk 1 above)
Comment (satoru-takeuchi) [reply cluster/examples/kubernetes/ceph/cluster.yaml]: This sentence disappeared after reflecting Travis's comment.

File: Documentation/ceph-cluster-crd.md [hunk 2]
```diff
@@ -1302,14 +1302,10 @@ spec:
 
 Rook has the ability to cleanup resources and data that were deployed when a `delete cephcluster` command is issued.
 The policy represents the confirmation that cluster data should be forcibly deleted.
-The cleanupPolicy should only be added to the cluster when the cluster is about to be deleted.
-After the `confirmation` field of the cleanup policy is set, Rook will stop configuring the cluster as if the cluster is about to be destroyed in order to prevent these settings from being deployed unintentionally.
 The `cleanupPolicy` CR settings has different fields:
```
[the remark this answers is not in the document]
Comment (satoru-takeuchi) [reply Documentation/ceph-cluster-crd.md]: Applied your additional commit. Thanks!

**Review by travisn — COMMENTED** [review]
You can squash the changes, no need to keep my suggestions separate. Thanks!

**Review by travisn — APPROVED** [review]

**Pull request merged**